src/libpmem/pmem.c

```diff
@@ -519,10 +519,12 @@ pmem_is_pmem(const void *addr, size_t len)
 	static int once;
 
 	/* This is not thread-safe, but pmem_is_pmem_init() is. */
+	VALGRIND_ANNOTATE_HAPPENS_AFTER(&once);
 	if (once == 0) {
 		pmem_is_pmem_init();
 		once++;
```

Comment:
I think this should be fetch_and_add or you should add a barrier after the once++;
I understand it's not threadsafe, but some threads might never notice the change to `once` technically speaking (as far as the C standard is concerned).